File: modules/algorithms-and-ds/modules/data-structures/linear/linked-list/src/python/singly-linked-list/driver_insert_using_SinglyLinkedList_class.py
from SinglyLinkedList import SinglyLinkedList
from Node import Node

sll = SinglyLinkedList()

# Assign data using constructor.
sll.head = Node(10)
sll.head.next = Node(20)
sll.head.next.next = Node(30)

# Assign data using "." operator.
# sll.head.data = 1
# sll.head.next.data = 2
# sll.head.next.next.data = 3

print("Value in the First Node (head):", sll.head.data)
print("Value in the Second Node:", sll.head.next.data)
print("Value in the Third Node (tail):", sll.head.next.next.data)

File: modules/algorithms-and-ds/modules/data-structures/linear/lists/src/cpp/singly-linked-list/driver_insert_using_SinglyLinkedList_class.cpp
#include "SinglyLinkedList.h"
#include <iostream>

int main()
{
    SinglyLinkedList sll;

    // Assign data using constructor.
    sll.head = new Node(10);
    sll.head->next = new Node(20);
    sll.head->next->next = new Node(30);

    // Assign data using "->" operator.
    // sll.head->data              = 10;
    // sll.head->next->data        = 20;
    // sll.head->next->next->data  = 30;

    std::cout << "Value in the First Node (head): " << sll.head->data << "\n";
    std::cout << "Value in the Second Node: " << sll.head->next->data << "\n";
    std::cout << "Value in the Third Node (tail): " << sll.head->next->next->data << "\n";

    return 0;
}
